openvdb/math/Vec3.h

```diff
@@ -265,79 +265,79 @@ class Vec3: public Tuple<3, T>
     template <typename S>
     const Vec3<T> &operator*=(S scalar)
     {
-        this->mm[0] = static_cast<T>(this->mm[0] * scalar);
-        this->mm[1] = static_cast<T>(this->mm[1] * scalar);
-        this->mm[2] = static_cast<T>(this->mm[2] * scalar);
+        this->mm[0] *= static_cast<T>(scalar);
+        this->mm[1] *= static_cast<T>(scalar);
+        this->mm[2] *= static_cast<T>(scalar);
         return *this;
```

Comment:
I've realised that changing casts like this will break cases where floats are applied to ints. I'm surprised that nothing in the unit tests has caught this. Trivial example. `Vec3i a(2); a *= 1.8f;`. a before would be equal to 3. Now, 1.8f will be truncated prior to multiplication. Silly of me to blanket change these. Thought here welcome

Reply:
Yes, arithmetic should be done wherever possible using the highest-precision operand's type (the `math::promote<>` metafunction is intended to help with that), with only the result being converted to the target type.

Also, the goal is not to suppress all type conversion warnings.  We want to see type conversion warnings, but we also want there to be at least one way to invoke any given operation without warnings if the user determines that the type conversions involved are safe for that operation.

Reply:
Yeah I agree, in which case I suspect the best solution would be to revert these vec3 changes and instead ensure that core vdb logic which invokes vec3 operators operates at the same/suitable precision. I can't remember how many warnings there were though, I'll take another look